src/main/java/com/zombiecastlerush/building/Shop.java

```diff
@@ -0,0 +1,48 @@
+package com.zombiecastlerush.building;
+
+import com.zombiecastlerush.entity.Player;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Shop extends Room {
```

Comment:
I realize that we never talked about what a Shop is, an interface with which you interact with in a room? Or a Room itself? I assumed former, it looks like you implemented latter, which seems workable now that I'm looking at it.